core/types/calls.cc

```diff
@@ -200,7 +200,7 @@ SymbolRef guessOverload(Context ctx, SymbolRef inClass, SymbolRef primary,
                         InlinedVector<const TypeAndOrigins *, 2> &args, const TypePtr &fullType, vector<TypePtr> &targs,
                         bool hasBlock) {
     counterInc("calls.overloaded_invocations");
-    ENFORCE(ctx.permitOverloadDefinitions(), "overload not permitted here");
+    ENFORCE(ctx.permitOverloadDefinitions(primary.data(ctx)->loc().file()), "overload not permitted here");
```

Comment:
Why not just `ctx.withOwner(primary).permitOverloadDefinitions()` instead? It looks like `permitOverloadDefinitions` is already consulting the owner.

Reply:
This would work here but not in the other uses of `permitOverloadDefinitions`, which runs earlier and therefore has less information available to it.